fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/IcebergColumnMinMaxMgr.java

```diff
@@ -0,0 +1,143 @@
+    private record CacheKey(ColumnIdentifier id, StatsVersion version) {}
+
+    private static final Logger LOG = LogManager.getLogger(IcebergColumnMinMaxMgr.class);
+
+    private static final SimpleExecutor stmtExecutor =
+            new SimpleExecutor("LakeColumnMinMaxMgr", TResultSinkType.HTTP_PROTOCAL);
+
+    private static final IcebergColumnMinMaxMgr INSTANCE = new IcebergColumnMinMaxMgr();
+
+    private final AsyncLoadingCache<CacheKey, Optional<ColumnMinMax>> cache = Caffeine.newBuilder()
+            .maximumSize(Config.statistic_dict_columns)
+            .executor(ThreadPoolManager.getStatsCacheThreadPoolForLake())
+            .buildAsync(new CacheLoader());
+
+    static IcebergColumnMinMaxMgr getInstance() {
+        return INSTANCE;
+    }
+
+    @Override
+    public Optional<ColumnMinMax> getStats(ColumnIdentifier identifier, StatsVersion version) {
+        if (!ConnectContext.get().getSessionVariable().isEnableMinMaxOptimization()) {
+            return Optional.empty();
+        }
+
+        try {
+            CompletableFuture<Optional<ColumnMinMax>> result = cache.get(new CacheKey(identifier, version));
+            if (result.isDone()) {
+                return result.get();
+            }
+        } catch (Exception e) {
+            LOG.warn("Failed to get MinMax for column: {}, version: {}", identifier, version, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public void removeStats(ColumnIdentifier identifier, StatsVersion version) {
+        throw new SemanticException("Iceberg column min/max stats unsupported the method");
+    }
+
+    @Override
+    public Map<String, Long> estimateCount() {
+        return Map.of("IcebergMinMax", (long) cache.asMap().size());
+    }
+
+    @Override
+    public List<Pair<List<Object>, Long>> getSamples() {
+        return List.of(Pair.create(List.of(new ColumnMinMax("1", "10000")), (long) cache.asMap().size()));
+    }
+
+    private static final class CacheLoader implements AsyncCacheLoader<CacheKey, Optional<ColumnMinMax>> {
+        @Override
+        public @NonNull CompletableFuture<Optional<ColumnMinMax>> asyncLoad(@NonNull CacheKey key,
+                                                                            @NonNull Executor executor) {
+            return CompletableFuture.supplyAsync(() -> {
+                try {
+                    List<String> names = StatisticsUtils.getTableNameByUUID(key.id.getTableUuid());
+                    if (names.size() < 3) {
+                        LOG.warn("tableUUID {} error for collecting min/max", key.id.getTableUuid());
+                        return Optional.empty();
+                    }
+
+                    String columnName = key.id.getColumnName().toSql(true);
+                    String tableName = StatisticUtils.quoting(names.get(0), names.get(1), names.get(2));
+
+                    String sql = "select min(" + columnName + ") as min, max(" + columnName + ") as max " +
```

Comment:
we can convert min/max iceberg query to metadata scan node now. you could check the plan. and please add some tests

Reply:
I think external table may be don't need support...